### ISSUE TITLE
Cloud function specific rate limit is not working

### ISSUE THREAD

**Issue by seyitdemirbas** [parse-community/parse-server#8684]
### New Issue Checklist

- [x ] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [x ] I am not just asking a [question](https://github.com/parse-community/.github/blob/master/SUPPORT.md).
- [x ] I have searched through [existing issues](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).
- [x] I can reproduce the issue with the [latest version of Parse Server](https://github.com/parse-community/parse-server/releases). 

### Issue Description

When I set rate limit in the cloud code, the rate limit is not working.

### Steps to reproduce
When I set options on parse server config the rate limit is working, like below code.

`rateLimit: {
    requestPath: '/functions/someFunction',
    requestTimeWindow: 15 * 60 * 1000,
    requestCount: 3,
  }`

but i'm trying to set rate limit in the cloud code, the rate limit is not working. like below code

`Parse.Cloud.define('someFunction', () => {
  return 'Hello world';
}, {
  rateLimit: {
    requestTimeWindow: 60 * 60 * 1000,
    requestCount: 3,
  }
});`

Server
- Parse Server version: `6.2.1`
- Operating system: `Ubuntu 20`
- Local or remote host (AWS, Azure, Google Cloud, Heroku, Digital Ocean, etc): `Local`

Database
- System (MongoDB or Postgres): `MongoDB`
- Database version: `6.0`
- Local or remote host (MongoDB Atlas, mLab, AWS, Azure, Google Cloud, etc): `Local`

Client
- SDK (iOS, Android, JavaScript, PHP, Unity, etc): `JavaScript`
- SDK version: `4.1.0`

**Comment by mtrezza** [comment]
> the rate limit is not working

Could you specify what you mean by that? How are you calling the Cloud Function? Mind that internal calls are ignored by the rate limit by default.

Could you try out an exact copy of the test case in your deployment?

https://github.com/parse-community/parse-server/blob/c9b59719ec9648f9ba8bbef89db2a24a227ae55c/spec/RateLimit.spec.js#L305-L317

**Comment by seyitdemirbas** [comment]
I am using Parse Server Example  to start parse server.

adding below code in cloud. 

Parse.Cloud.define('test', () => 'Abc', {
    rateLimit: {
      requestTimeWindow: 15 * 60 * 1000,
      requestCount: 1,
      includeInternalRequests: true,
    },
});

and using parse dashboard REST console POST on functions/test
result;
{
    "result": "Abc"
}

sending more POST request and rate limit is not working. always i am get result;
{
    "result": "Abc"
}


now i am adding index.js which parse server config, this code 

`rateLimit: {
    requestPath: '/functions/test',
    requestTimeWindow: 15 * 60 * 1000,
    requestCount: 1,
    includeInternalRequests: true
 },`

and i used parse dashboard REST console POST on functions/test
{
    "result": "Abc"
}

second try;
{
    "code": 100,
    "error": "Too many requests."
}

this time the rate limit is working properly

on cloud rate limiter is not working. i think this is a bug.

**Comment by mtrezza** [comment]
Thanks for the details.

**Comment by Moumouls** [comment]
@syt5512 on the cloud i suppose that you have some horizontal scaling ?

**Comment by seyitdemirbas** [comment]
> @syt5512 on the cloud i suppose that you have some horizontal scaling ?

No, i am using my computer, working on localhost, just one operating system.
